src/controller/java/templates/CHIPClusters-JNI.zapt

```diff
@@ -70,12 +70,15 @@ JNI_METHOD(void, {{asUpperCamelCase ../name}}Cluster, {{asLowerCamelCase name}})
     auto successFn = chip::Callback::Callback<CHIP{{>callbackName}}CallbackType>::FromCancelable(onSuccess->Cancel());
     auto failureFn = chip::Callback::Callback<CHIPDefaultFailureCallbackType>::FromCancelable(onFailure->Cancel());
 
-    err = cppCluster->InvokeCommand(request, onSuccess->mContext, successFn->mCall, failureFn->mCall
     {{#if mustUseTimedInvoke}}
-      {{!TODO Fix Java API to pass in this information.  For now, 10 seconds.}}
-      , 10000
+    err = cppCluster->InvokeCommand(request, onSuccess->mContext, successFn->mCall, failureFn->mCall, chip::JniReferences::GetInstance().IntegerToPrimitive(timedInvokeTimeoutMs));
```

Comment:
What will this do if `timedInvokeTimeoutMs == nullptr`?  Or can that not happen with an actual call from Java?

Reply:
It won't be nullptr because of how the Java code is generated. If `mustUseTimedInvoke = true`, the only Java method available will always have the argument `int timedInvokeTimeoutMs`, which can't be null. The same thing has been set up for writes.